pkg/registry/apis/provisioning/read.go

```diff
@@ -0,0 +1,107 @@
+package provisioning
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/registry/rest"
+
+	common "github.com/grafana/grafana/pkg/apimachinery/apis/common/v0alpha1"
+	provisioning "github.com/grafana/grafana/pkg/apis/provisioning/v0alpha1"
+)
+
+type readConnector struct {
+	getter rest.Getter
+}
+
+func (*readConnector) New() runtime.Object {
+	// This is added as the "ResponseType" regardless what ProducesObject() returns
+	return &provisioning.ResourceWrapper{}
+}
+
+func (*readConnector) Destroy() {}
+
+func (*readConnector) NamespaceScoped() bool {
+	return true
+}
+
+func (*readConnector) GetSingularName() string {
+	return "Resource"
+}
+
+func (*readConnector) ProducesMIMETypes(verb string) []string {
+	return []string{"application/json"}
+}
+
+func (*readConnector) ProducesObject(verb string) any {
+	return &provisioning.ResourceWrapper{}
+}
+
+func (*readConnector) ConnectMethods() []string {
+	return []string{http.MethodGet}
+}
+
+func (*readConnector) NewConnectOptions() (runtime.Object, bool, string) {
+	return nil, true, "" // true adds the {path} component
+}
+
+func (s *readConnector) Connect(ctx context.Context, name string, opts runtime.Object, responder rest.Responder) (http.Handler, error) {
+	obj, err := s.getter.Get(ctx, name, &metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	repo, ok := obj.(*provisioning.Repository)
+	if !ok {
+		return nil, fmt.Errorf("expected repository, but got %t", obj)
+	}
+
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		idx := strings.Index(r.URL.Path, "/"+name+"/read")
+		path := strings.TrimLeft(r.URL.Path[idx+len(name+"/read")+2:], "/")
+		if path == "" {
+			responder.Error(fmt.Errorf("missing path")) // TODO bad request
+			return
+		}
+		commit := r.URL.Query().Get("commit")
+
+		// TODO, actually read, and then validate
+		obj := &unstructured.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "dashboards.grafana.app/v????",
+				"metadata": metav1.ObjectMeta{
+					Name: "xxx",
+				},
+				"spec": map[string]interface{}{
+					"path":   path,
+					"commit": commit,
+					"repo":   repo.Name,
+					"type":   repo.Spec.Type,
+				},
+			},
```

Comment:
(posting even though it's inactionable, for others' sake)
I was going to ask: Do we want to reimplement the dashboard model here?
But then I looked at the dashboard model:

```go
// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
type Dashboard struct {
	metav1.TypeMeta `json:",inline"`
	// Standard object's metadata
	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
	// +optional
	metav1.ObjectMeta `json:"metadata,omitempty"`

	// The dashboard body (unstructured for now)
	Spec common.Unstructured `json:"spec"`
}
```

I guess this is alright 😂

Reply:
For this endpoint the nested resource is whatever type we read -- it could be `dashboards.grafana.app/...` or  `playlists.grafana.app/...` etc

As this gets real -- the endpoint could look at the value and then validate against that api, before returning the result